### PR TITLE
Guido/ab8704 convert csma to schema based table creation

### PR DESCRIPTION
**Pull request by gkoller**
Use schema to create DB tables.

Lot's of additional cleanups and fixes. Best reviewed per commit.

  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Use PostgresRenameOperator instead of literal SQL.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Fixed MyPy errors for PostgresRenameOperator.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Don't return the super call to execute.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Add CASCADE support to rename operator.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Add pyugrade cleanups commit to .git-blame-ignore-revs.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Add pyupgrade to .pre-commit and as dev dependency.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Fix: deprecated import & pass execution_date to super().
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Some pyugrade code clean-ups.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Conditionally drop identity.
  * Add .git-blame-ignore-revs file
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Schema based table creation for cmsa.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Use upper case names for constants.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Enable py38 specific behavior. Replace print w. logging.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Airflow uses pendulum. So should custom hooks.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Configure LoginMixin by calling super().__init__().
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Add Path support to SwiftHook.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Do not hard code timezone.
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Fix: Seriously? ntpath instead of os.path on Unix?
  * [AB#8704](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8704) Use new @task mk_dir instead of BashOperator.

When rolled out to acceptance and/or production we should run the following queries on the respective databases:

```sql
DROP SEQUENCE IF EXISTS cmsa_locatie_id_seq CASCADE;
DROP SEQUENCE IF EXISTS cmsa_locatie_new_id_seq CASCADE;
DROP SEQUENCE IF EXISTS cmsa_locatie_old_seq CASCADE;
DROP SEQUENCE IF EXISTS cmsa_locatie_id_seq1 CASCADE;
DROP SEQUENCE IF EXISTS cmsa_locatie_new_id_seq1 CASCADE;
DROP SEQUENCE IF EXISTS cmsa_locatie_old_seq1 CASCADE;
```
This is due to the change from a `serial` based column to an `identity` based one. Didn't make any sense in coding that into to DAG as it is only used once, namely in this transition to schema based table creation.

### CODE REVIEW

File: src/dags/aardgasvrijezones.py [hunk 1]
```diff
@@ -49,7 +51,7 @@
     )
 
     # 2. Create temp directory to store files
-    mkdir = BashOperator(task_id="mkdir", bash_command=f"mkdir -p {tmp_dir}")
+    mkdir = mk_dir(Path(tmp_dir))
```
Comment (cvriel) [inline src/dags/aardgasvrijezones.py:54]: :+1: nice to see this applied to all dags

**Review by cvriel — APPROVED** [review]

File: .git-blame-ignore-revs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# See: https://akrabat.com/ignoring-revisions-with-git-blame/
+#
+# List commit hashes (full 40 characters) of commits to be ignore when using
+# git blame. We generally want to exclude commits that soley formatted parts or
```
Comment (lbam) [inline .git-blame-ignore-revs:4]: soleLy

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# the entire code base (another reason to use atomic commits). In order for
+# this file to be taken into consideration by git run the following command:
+#
+# git config blame.ignoreRevsFile .git-blame-ignore-revs
+#
+# To update this file, simple _append_ a commit to it including the first line
```
Comment (lbam) [inline .git-blame-ignore-revs:10]:
```suggestion
# To update this file, simply _append_ a commit to it, including the first line
```

File: src/dags/bedrijveninvesteringszone.py [hunk 4]
```diff
@@ -52,7 +54,7 @@
     )
 
     # 2. Create temp directory to store files
-    mkdir = BashOperator(task_id="mkdir", bash_command=f"mkdir -p {tmp_dir}")
+    mkdir = mk_dir(Path(tmp_dir))
```
Comment (lbam) [inline src/dags/bedrijveninvesteringszone.py:57]: Looks like we could get rid of some noise by having `mk_dir` work on `Union[Path, str]`...

Comment (lbam) [reply src/dags/bedrijveninvesteringszone.py:57]: .... except that some paths are still constructed using string formatting. Never mind, let's tackle this in a follow-up.

File: .git-blame-ignore-revs (hunk 2 above)
Comment (gkoller) [reply .git-blame-ignore-revs]: Thanks, fixed.

File: .git-blame-ignore-revs (hunk 3 above)
Comment (gkoller) [reply .git-blame-ignore-revs]: Thanks, fixed.

File: src/plugins/dynamic_dagrun_operator.py [hunk 5]
```diff
@@ -1,10 +1,8 @@
-from __future__ import annotations
-
 from datetime import datetime
-from typing import Any, Optional
+from typing import Any, Dict, Optional
 
 from airflow.models.dag import DagModel, DagRun
-from airflow.operators.dagrun_operator import TriggerDagRunOperator
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
```
Comment (vdboor) [inline src/plugins/dynamic_dagrun_operator.py:5]: heeft pyupgrade dit ook gedaan?

File: src/plugins/dynamic_dagrun_operator.py [hunk 6]
```diff
@@ -46,14 +44,14 @@ def __init__(
         """
         super().__init__(
             trigger_dag_id=None,
-            execution_date=None,
+            execution_date=execution_date,
```
Comment (vdboor) [inline src/plugins/dynamic_dagrun_operator.py:47]: handig ;-)

**Review by vdboor — APPROVED** [review]

File: src/plugins/dynamic_dagrun_operator.py (hunk 5 above)
Comment (gkoller) [reply src/plugins/dynamic_dagrun_operator.py:5]: Nope, that was me. For some strange reason it changed `Optional[datetime]` into `datetime | None` for this file even though I ran it with `--py38-plus`. So I had to revert it manually. While doing so I saw the deprecated import and fixed that immediately. Should have been a separated commit.

File: src/plugins/dynamic_dagrun_operator.py (hunk 6 above)
Comment (gkoller) [reply src/plugins/dynamic_dagrun_operator.py:47]: Another manual change that should have been a separate commit. Let me see if I can change that.

File: src/dags/bedrijveninvesteringszone.py (hunk 4 above)
Comment (gkoller) [reply src/dags/bedrijveninvesteringszone.py:57]: Good point. Though I have given this some thought. 

`mk_dir` is a brand new function. For anything new I'd like to use `pathlib` exclusively. It is different when I want to add `pathlib` capabilities to an existing function; then I do go from `str` to `Union[Path, str]`.

By using `pathlib` exclusively for new functions. Their call sites stand out with the explicit `Path` cast. That makes it easier to address `str` versus `Path` usage at those call sites.

File: src/plugins/dynamic_dagrun_operator.py (hunk 5 above)
Comment (gkoller) [reply src/plugins/dynamic_dagrun_operator.py:5]: Done.

File: src/plugins/dynamic_dagrun_operator.py (hunk 6 above)
Comment (gkoller) [reply src/plugins/dynamic_dagrun_operator.py:47]: Done.

**Pull request merged**